tensorboard/backend/security_validator.py

```diff
@@ -27,29 +27,44 @@
 
 from tensorboard.util import tb_logging
 
-Policy = collections.namedtuple("Poicy", ["name", "rules"])
+# Loosely follow vocabulary from https://www.w3.org/TR/CSP/#framework-directives.
+Directive = collections.namedtuple("Directive", ["name", "value"])
 
 logger = tb_logging.get_logger()
 
 _HTML_MIME_TYPE = "text/html"
-_CSP_PATTERN = re.compile(r"^([a-z]+-[a-z]+)(.*)")
+# Matches CSP rule like "strict-src foobar baz".
+_CSP_PATTERN = re.compile(r"^([a-zA-Z0-9-]+) (.*)")
```

Comment:
Strictly speaking the space following the directive name shouldn't be required if the value is empty (for directives with no value), so this is also a bit of a deviation since it always requires the space.

Reply:
I instead decided to use `split`. PTAL.